src/JsonSchema/Constraints/TypeConstraint.php

```diff
@@ -185,6 +185,10 @@ protected function validateType(&$value, $type, $coerce = false)
         }
 
         if ('array' === $type) {
+            if ($coerce) {
+                $value = $this->toArray($value);
```

Comment:
Looks good, but I notice that ajv only does this conversion if `coerceTypes` is set to `array`. Should this be controlled by another flag?